producer-kafka/src/main/java/io/stargate/producer/kafka/KafkaCDCProducer.java

```diff
@@ -48,10 +51,20 @@
 
   private CompletableFuture<CompletableKafkaProducer<GenericRecord, GenericRecord>> kafkaProducer;
 
-  public KafkaCDCProducer() {
+  public KafkaCDCProducer(MetricRegistry registry) {
+    registerMetrics(registry);
     this.configLoader = new DefaultConfigLoader();
   }
 
+  /**
+   * It registers the provided MetricRegistry in the Dropwizard shared metrics registry (see {@link
+   * SharedMetricRegistries}). The {@link DropwizardMetricsReporter} is getting the metrics registry
+   * via {@code SharedMetricRegistries.getOrCreate("default")} in the constructor.
+   */
+  private void registerMetrics(MetricRegistry registry) {
+    SharedMetricRegistries.add("default", registry);
```

Comment:
Is using `"default"` key the same as using `setDefault()` ?

Reply:
the `setDefault()` have a different semantic. The `add()` is just adding a registry with a given name. The `default` name is used only because the `DropwizardMetricsReporter` is using it as well. It can be any arbitrary name.
The `setDefault` is also setting the default registry, but it fails if it is already set, and sets the name of a new default globally.